agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -231,19 +352,14 @@ private static SpanExporter createSpanExporter(
               }
               return false;
             },
-            telemetryItems -> {
-              for (TelemetryItem telemetryItem : telemetryItems) {
-                telemetryClient.trackAsync(telemetryItem);
-              }
-              return CompletableResultCode.ofSuccess();
-            },
-            CompletableResultCode::ofSuccess,
-            CompletableResultCode::ofSuccess,
-            () -> !Strings.isNullOrEmpty(TelemetryClient.getActive().getInstrumentationKey()),
             AiAppId::getAppId);
 
+    BatchItemProcessor batchItemProcessor = telemetryClient.getGeneralBatchItemProcessor();
+
     SpanExporter spanExporter =
-        new StatsbeatSpanExporter(azureMonitorTraceExporter, telemetryClient.getStatsbeatModule());
+        new StatsbeatSpanExporter(
+            new AgentSpanExporter(mapper, quickPulse, batchItemProcessor),
```

Comment:
why does quickPulse has anything to do with Statsbeat?

Reply:
it doesn't, the StatsbeatSpanExporter comes first in the SpanExporter chain before the AgentSpanExporter